ras/ServiceReport.py

```diff
@@ -49,8 +49,62 @@ def setUp(self):
         self.sourcedir = os.path.join(self.workdir, 'ServiceReport-master')
         build.make(self.sourcedir)
 
+    def spyrePresent(self):
+        for dev in os.listdir('/sys/bus/pci/devices'):
+            cls = os.path.join('/sys/bus/pci/devices', dev, 'class')
+            try:
+                with open(cls) as f:
+                    if f.read().strip().startswith('0x1200'):
+                        return True
+            except Exception:
+                pass
+        return False
+
     def test(self):
         os.chdir(self.sourcedir)
+
+        if not self.spyrePresent():
+            self.cancel("Spyre Accelerator not present on this system")
```

Comment:
this test will be used for non spyre  use case... so do not cancel  instead move all spyre related options/test inside the if spyrePresent() block